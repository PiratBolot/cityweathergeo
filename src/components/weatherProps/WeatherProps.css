.wn_body {
    padding: 0 10px;
    flex-grow: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.wn_line {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 5px 10px;
    color: white;
    background-color: #8199A3;
    margin: 5px 0;
}

.info_label {
    font-weight: bold;
    margin-right: 20px;
}

.info_value {
    font-weight: bold;
    color: #435257;
}