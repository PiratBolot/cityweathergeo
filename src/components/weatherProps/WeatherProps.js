import React from "react";
import './WeatherProps.css'

class WeatherProps extends React.Component {
    render() {
        return (
        <div className="wn_body">
            {
                this.props.parsedWeather.map((e) =>
                    <div key={e.key} className="wn_line">
                        <div className="info_label">{e.key}</div>
                        <div className="info_value">{e.value}</div>
                    </div>
                )
            }
        </div>
        )
    }
}

export default WeatherProps;