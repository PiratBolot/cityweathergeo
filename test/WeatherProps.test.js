import React from 'react';
import { shallow } from '../enzyme/enzyme';
import { shallowToJson } from 'enzyme-to-json';

import WeatherProps from "../src/components/weatherProps/WeatherProps";
import parsedWeatherProps from "./inputDataExample/parsedWeatherProps";

describe('Weather Properties', () => {
    const parsedWeatherExample = parsedWeatherProps();

    it('should render correctly', () => {
        const output = shallow(
            <WeatherProps parsedWeather={parsedWeatherExample}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});

