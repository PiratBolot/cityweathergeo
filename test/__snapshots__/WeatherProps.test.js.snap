// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Properties should render correctly 1`] = `
<div
  className="wn_body"
>
  <div
    className="wn_line"
    key="Ветер"
  >
    <div
      className="info_label"
    >
      Ветер
    </div>
    <div
      className="info_value"
    >
      3 м/с, Ю-Ю-З
    </div>
  </div>
  <div
    className="wn_line"
    key="Давление"
  >
    <div
      className="info_label"
    >
      Давление
    </div>
    <div
      className="info_value"
    >
      1005 hPa
    </div>
  </div>
  <div
    className="wn_line"
    key="Влажность"
  >
    <div
      className="info_label"
    >
      Влажность
    </div>
    <div
      className="info_value"
    >
      86 %
    </div>
  </div>
  <div
    className="wn_line"
    key="Координаты"
  >
    <div
      className="info_label"
    >
      Координаты
    </div>
    <div
      className="info_value"
    >
      [61.67, 50.82]
    </div>
  </div>
</div>
`;
